tests/test_acceptance.py

```diff
@@ -51,6 +51,7 @@ def test_reader_all_primitive_types():
     # timestamp differently. This is likely a bug in arrow but the delta result
     # is "more correct".
     expected_ddf["timestamp"] = expected_ddf["timestamp"].astype("datetime64[us]")
+    expected_ddf["timestamp"] = expected_ddf["timestamp"].dt.tz_localize("UTC")
```

Comment:
I've opted to continue to manually match what delta is returning as this is what we've already been doing. It'd be good to track down what the underlying issue here, but that seems like a separate issue from the changes in this PR. 

Searching around a bit, it looks like there are known timestamp issues in delta-rs https://github.com/delta-io/delta-rs/milestone/5 and the corresponding `all_primitive_types` upstream is [being skipped](https://github.com/delta-io/delta-rs/blob/b001186d27b9ae8bfee9f40a1a4be88afb9fa0e1/python/tests/data_acceptance/test_reader.py#L49)